### DISCUSSION TITLE
Mosquitto passwd

### DISCUSSION THREAD

**Discussion by andreadannibale** [indomus/forum#571] [HUB: Home Assistant]
Salve a tutti.. utilizzo home assistant core installato su docker tramite docker-compose..docker gira su raspbian.. 

Su HA ho già integrato diversi shelly tramite mqtt.. oggi ne ho integrato uno nuovo.. impostato ip del broker sulla configurazione dello shelly..impostato nome utente e password.. tutto come gli altri shelly...

Il problema è che guardando il log di mosquitto.. questo shelly non si aggancia..non pubblica nulla..

Aggiungo però un altro aspetto... la configurazione di mosquitto l'ho messa a allow_anonymous True perchè ogni volta che aggiornavo mosquitto dovevo ricreare da capo l'utente... altrimenti nel log mi dava:

1627473975: Config loaded from /mosquitto/config/mosquitto.conf.

1627473975: Error: Unable to open pwfile "/mosquitto/config/pwfile".

1627473975: Error opening password file "/mosquitto/config/pwfile"

Non essendo riuscito a capire l'origine del problema, ho cambiato in allow_anonymous true..

**Comment by snakuzzo** [comment]
Per quanto riguarda lo Shelly, se non si connette e se sei sicuro di aver impostato correttamente tutti i parametri, verifica il nome che hai utilizzato sul device...intendo quello utilizzato nel topic MQTT. Se è troppo lungo potrebbe creare problemi.

Quanto al discorso file e utente di MQTT...ti posso assicurare al 101% che hai sbagliato montare i volumi (directory e file di configurazione).
Quindi prima di tutto ti consiglierei di ridare un'occhiata all'installazione.

**Reply by andreadannibale** [reply]
per lo shelly il nome è semplicissimo (lampada)...per l'altro discorso relativo all'installazione di mosquitto ho questo:
``` volumes:
    - "${PWD}/mosquitto/config:/mosquitto/config" 
 ```

Quindi potrebbe essere un problema di permessi della cartella config?

**Reply by andreadannibale** [reply]
Ho risolto il problema dei volumi riguardando la guida indomus.. grazie snakuzzo.. lo shelly invece non sono ancora riuscito.. risulta connesso alla rete di casa.. risulta impostato mqtt.. disconnesso dal cloud..tutti i parametri ok... non pubblica ancora nulla..

**Reply by andreadannibale** [reply]
Risolto, non so per quale motivo ma pochi minuti fa l'ho finalmente trovato nella coda mqtt..
